enhancements/20240604-lvm-csi-driver.md

```diff
@@ -0,0 +1,77 @@
+# LVM CSI Driver
+
+## Summary
+
+In some specific scenarios, we need better performance and could use a strict location for the volume. The LVM could be a good choice because it is a thin layer on top of the block devices. Also, it provides useful features like volume expansion and snapshots. We want to introduce the LVM CSI driver to support the LVM volume in Harvester. 
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/5724
+
+## Motivation
+
+We do not have any current solution to support local volumes, which means directly using the local disk as the volume. The LVM CSI driver could help leverage the local disk as the volume. If the user could use the strict location VM, it would help us improve performance and latency.
+
+### Goals
+
+Introduce the LVM CSI driver to support the LVM volume in Harvester.
+
+### Non-goals [optional]
+
+- Support volume capability and snapshot for the LVM volume.
+- Support downstream cluster to use the LVM volume.
+
+## Proposal
+
+We would like to start with the existing LVM CSI driver and introduce it to the Harvester. We need to make some changes to make it more suitable for the Harvester. After that, we could use the LVM CSI driver to create the LVM volume in Harvester.
+
+### User Stories
+
+#### The user creates the PCI pass-through VM and wants a higher performance.
+When users use the features with PCI pass-through, it means they will redirect the VM to a specific host. Also, the user wants a higher-performance disk for the VM. The LVM CSI driver could help users create the PVC with an LVM volume on a local disk. That could help users have better performance and latency.
+
```

Comment:
Is PCI pass-through required? The description in https://github.com/harvester/harvester/issues/5724 just says "Applications require large volume or high-performance local disks. Live-migration is not a concern." AFAICT it would be enough to create a VM with Node Scheduling set to "Run VM on specific node - (Live migration is not supported", then specify the node to run on, matching the node with the LVM volume.

Reply:
Thanks for the suggestion. I rewrote this part for the feature with PCI passthrough.

But for the Detail experience, I still use that as an example. WDYT?